packages/mui-base/src/Unstable_NumberInput/NumberInput.test.tsx

```diff
@@ -498,32 +498,64 @@ describe('<NumberInput />', () => {
       expect(input.value).to.equal('1');
     });
 
-    it('sets value to min when the input has no value and ArrowUp is pressed', async () => {
-      const handleChange = spy();
+    describe('when the input has no value and ArrowUp is pressed', () => {
+      it('sets value to min if min is provided', async () => {
```

Comment:
This isn't quite how the native control works - it kinda assumes that no value = 0 (so if `min = 0`, pressing the ArrowUp sets the control to `1`.

Reply:
@michaldudak This was intentional ~ I think it's a worthwhile improvement from the native behavior 😬 

If a range is defined with `min` and/or `max` in a cleared NumberInput, it feels like a  better UX to start incrementing from `min` on arrow up, and start decrementing from the `max` on arrow down

If arrow down starts on the `min` value, a 2nd arrow down will do nothing, and I feel its more natural to tap the same arrow 2-3 times – e.g. increment from `min` to `min+3`, or decrement from `max` to `max-3` – rather than "down-up-up-up"

Did a quick check as well: react-aria/react-spectrum also implemented this behavior – [sandbox](https://codesandbox.io/p/sandbox/frosty-ritchie-2wgm58?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clqwff35c00063b6d1wuql1m0%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clqwff35c00023b6dx9mq36l4%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clqwff35c00033b6d82x9jro3%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clqwff35c00053b6daj41tjhh%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clqwff35c00023b6dx9mq36l4%2522%253A%257B%2522id%2522%253A%2522clqwff35c00023b6dx9mq36l4%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clqwff35c00053b6daj41tjhh%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqwff35c00043b6d5twc0kn7%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clqwff35c00053b6daj41tjhh%2522%252C%2522activeTabId%2522%253A%2522clqwff35c00043b6d5twc0kn7%2522%257D%252C%2522clqwff35c00033b6d82x9jro3%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clqwff35c00033b6d82x9jro3%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) - but mantine and Chakra do not

Reply:
BTW I would love to get @colmtuite's opinion on this as well ~

Reply:
Yes I agree @mj12albert. When `min` or `max` is set, it should increment/decrement from those values imo.

Reply:
All right, then. Let's get it merged!